docs/src/man/man3/hal_stream.3hal.adoc

```diff
@@ -65,80 +61,48 @@ stream was attached with *hal_stream_attach* rather than created with
 
 === *hal_stream_attach*
 
-Attach the given stream, which was already created by
-*hal_stream_create*. If the typestring is specified, this call fails if
-it does not match the typestring the stream was created with. If the
-typestring argument is NULL, then any typestring is accepted.
+Attach the given stream, which was already created by *hal_stream_create*.
+If the typestring is specified, this call fails if it does not match the typestring the stream was created with.
+If the typestring argument is NULL, then any typestring is accepted.
 
 === *hal_stream_detach*
 
 Detach the given stream. It is an undiagnosed error if the stream was
-created with *hal_stream_create* rather than attached with
-*hal_stream_attach*. It is an undiagnosed error if the call to
-*hal_stream_detach* is omitted.
+created with *hal_stream_create* rather than attached with *hal_stream_attach*.
+It is an undiagnosed error if the call to *hal_stream_detach* is omitted.
 
-=== *hal_stream_element_count*
+*hal_stream_element_count*:: Returns the number of pins.
 
-Returns the number of pins.
+*hal_stream_element_type*:: Returns the type of the given pin number.
 
-=== *hal_stream_element_type*
+*hal_stream_readable*:: Returns true if the stream has at least one sample to read
 
-Returns the type of the given pin number.
+*hal_stream_read*:: If the stream has one sample to read, stores it in buf.
 
-=== *hal_stream_readable*
+*hal_stream_writable*:: Returns true if the stream has room for at least one sample to be written.
 
-Returns true if the stream has at least one sample to read
+*hal_stream_depth*:: Returns the number of samples waiting to be read.
 
-=== *hal_stream_read*
+*hal_stream_maxdepth*:: Returns the *depth* argument that the stream was created with.
 
-If the stream has one sample to read, stores it in buf.
+*hal_stream_num_overruns*:: Returns a number which is incremented each time *hal_stream_write* is called without space available.
 
-=== *hal_stream_writable*
+*hal_stream_num_underruns*:: Returns a number which is incremented each time *hal_stream_read* is called without a sample available.
 
-Returns true if the stream has room for at least one sample to be
-written.
+*hal_stream_wait_readable*:: Waits until the stream is readable or the stop flag is set.
 
-=== *hal_stream_depth*
+*hal_stream_wait_writable*:: Waits until the stream is writable or the stop flag is set.
 
-Returns the number of samples waiting to be read.
+*hal_stream_read*:: Reads a record from stream.
+  If successful, it is stored in the given buffer.
+  Optionally, the sample number can be retrieved. If no sample is available, _num_underruns_ is incremented.
+  It is an undetected error if more than one component or real-time function calls *hal_stream_read* concurrently.
 
-=== *hal_stream_maxdepth*
```

Comment:
What was wrong with this formatting? These changes makes this document a bit inconsistent.
Githubs diff is a bit bad at this point, VSCode shows it better:

![grafik](https://github.com/user-attachments/assets/fcda64f5-a0a7-4c13-bb42-ed50c9a1f3c6)

Reply:
It is incomplete, I agree. The "==="-lines above should also have been changed. All the other man pages have used the :: construct to explain their pins. And so should this man page, I had thought. The trigger to touch it at all were warnings during the build. What is different here is that the descriptions are so long, but not too long for the :: ?!?

Reply:
Not too ling for the :: IMHO.

Reply:
Nice, thanks